fgrpc/grpcrunner.go

```diff
@@ -83,6 +86,23 @@ type GRPCRunnerResults struct {
 	Destination string
 	Streams     int
 	Ping        bool
+	//Message header that received from grpc server
+	ReceivedMessageHeaderSize uint64
+	//Message header size that sent to grpc server
+	SentMessageHeaderSize uint64
+	//Message body size that sent to grpc server
+	SentMessageSize uint64
+}
+
+// GetRequestSize returns the request body size. If the request is Ping, Ping Message size will return. Otherwise
+// HealthCheckRequest's service len will return.
+func (grpcstate *GRPCRunnerResults) GetRequestSize() int {
+	if grpcstate.Ping {
+		size := int(unsafe.Sizeof(&grpcstate.reqP))
+		size += len(grpcstate.reqP.Payload)
+		return size
+	}
+	return len(grpcstate.reqH.Service)
```

Comment:
I doubt that's the one the wire size of the packet send, there is probably some header etc

also there should be a way to get that request size or bytes from grpc ? did you search for it ?

Reply:
I didn't have time to look in a detailed way. But I will hopefully take a look in the weekend.